staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -274,6 +280,15 @@ func (h *kmsv2PluginProbe) check(ctx context.Context) error {
 	return nil
 }
 
+// getCurrentKeyID returns the latest keyID from the Status() method or err if keyID is empty
+func (h *kmsv2PluginProbe) getCurrentKeyID(ctx context.Context) (string, error) {
```

Comment:
@liggitt this PR makes the API server coast (forever) on the last successful key ID it got from the plugin.  Thus staleness checks for read calls will not fail if the plugin is down transiently.  At a high level I believe we have three choices here:

1. Coast on the last key ID, no error, staleness based on last observed key ID
2. Error and fail the read request (would partially defeat the purpose of having a cache since we would no longer tolerate plugin downtime)
3. Mark the read request as `stale=true` - for no-op updates this would cause a write (which would presumably fail since the plugin is down)

What do you think is the best approach here?

Reply:
Maybe another option is to make `stale` a `*bool` to try and express the "I do not know" state?

Reply:
it's hard to imagine life being better for a cluster admin if we make the cluster start failing reads ... I wouldn't do that.

failing no-op writes doesn't seem useful to me either... it pushes errors to users unlikely to be able to do anything about them

I think I would expect to coast, and for an error here to increment a metric to make it clear something is not working as expected (either transiently or persistently)

Reply:
Opened https://github.com/kubernetes/kubernetes/issues/115188 to add metrics as a follow up.